File: frontend/src/app/Models/student.ts
export interface StudentModel {
  id?: number;
  FirstName: string;
  LastName: string;
  Grade: string;
  EnrollmentDate?: number;
}
export class StudentClass implements StudentModel {
  id?: number;
  FirstName: string;
  LastName: string;
  Grade: string;
  EnrollmentDate?: number;

  constructor(data: StudentModel) {
    Object.assign(this, data);
    return this;
  }
}

File: studentsCrud/Models/Student.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace studentsCrud.Models
{
    public class Student
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        [Required]
        public string Grade { get; set; }
        [Required]
        public string EnrollmentDate { get; set; }

    }
}